app/frontend/src/features/connections/friends/useFriendList.test.tsx

```diff
@@ -195,8 +166,61 @@ describe("when the listFriends query failed", () => {
       isLoading: false,
       isError: true,
       errors: ["Error listing friends"],
-      friendQueries: [],
+      data: undefined,
     });
     expect(getUserMock).not.toHaveBeenCalled();
   });
 });
+
+describe("with cached user data", () => {
+  it("returns isError as true with the stale data if subsequent refetch queries fail", async () => {
+    const sharedClient = new QueryClient({
+      defaultOptions: { queries: { retry: false } },
+    });
+    const sharedClientWrapper = ({
+      children,
+    }: {
+      children: React.ReactNode;
+    }) => (
+      <QueryClientProvider client={sharedClient}>
+        {children}
+      </QueryClientProvider>
+    );
+    renderHook(() => useFriendList(), {
+      wrapper: sharedClientWrapper,
+    });
+    await wait(0);
+
+    listFriendsMock.mockRejectedValue(new Error("Error listing friends"));
+    getUserMock.mockRejectedValue(new Error("Error fetching user data"));
+
+    const { result, waitForNextUpdate } = renderHook(() => useFriendList(), {
+      wrapper: sharedClientWrapper,
+    });
+    await waitForNextUpdate();
+
+    expect(result.current).toMatchObject({
+      isLoading: false,
+      isError: true,
+      errors: [
+        "Error listing friends",
+        "Error fetching user data",
+        "Error fetching user data",
+      ],
```

Comment:
Hmm this means if the user's connection drops and they have 20 friends and they try to load the friends list, they will get an alert box with 21 lines. What do you think? Should we just have one "error" per hook and print individual errors to the console with `console.error()`?

Reply:
Could save for another pr if you want.

Reply:
Yeah, I think I'll leave this for when we might refactor this hook to use your `useUsers` hook